packages/core/src/domain/error/trackRuntimeError.spec.ts

```diff
@@ -1,48 +1,40 @@
-import { disableJasmineUncaughtExceptionTracking, collectAsyncCalls, registerCleanupTask } from '../../../test'
+import { disableJasmineUncaughtExceptionTracking, wait } from '../../../test'
 import { Observable } from '../../tools/observable'
 import { isIE } from '../../tools/utils/browserDetection'
-import type { UnhandledErrorCallback } from './trackRuntimeError'
 import { instrumentOnError, instrumentUnhandledRejection, trackRuntimeError } from './trackRuntimeError'
 import type { RawError } from './error.types'
 
 describe('trackRuntimeError', () => {
   const ERROR_MESSAGE = 'foo'
 
-  let originalOnErrorHandler: OnErrorEventHandler
-  let onUnhandledrejectionSpy: jasmine.Spy
-  let onErrorSpy: jasmine.Spy
-  let notifyError: jasmine.Spy
-  let stopRuntimeErrorTracking: () => void
-
-  beforeEach(() => {
-    originalOnErrorHandler = window.onerror
-    onErrorSpy = jasmine.createSpy()
-    window.onerror = onErrorSpy
-
-    onUnhandledrejectionSpy = setupOnUnhandledrejectionSpy()
+  const withTrackRuntimeError = async (callback: () => void): Promise<RawError> => {
```

Comment:
The tests in this file involve some async steps, and getting the ordering right is tricky. I've reworked them with an `async` callback-based approach that (IMO) feels a bit cleaner and makes it easier to tell that things happen in the right order. I know this is a bit of a different pattern than most of the other tests, though!

Reply:
Now that I've rebased against v6, the delta is a bit smaller here, because these tests already used async/await in the v6 branch. The difference is now just that the flow of events in the tests is managed more explicitly, without the use of `collectAsyncCalls`.